src/components/common/header/AuthenticatedHeader.js

```diff
@@ -1,23 +1,30 @@
-import React, { Fragment, useEffect } from 'react';
-import { Link } from 'react-router-dom';
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { logoutAccount } from '@Redux/actions/authActions';
+import { connect } from 'react-redux';
+import { Link, withRouter } from 'react-router-dom';
 import Materialize from 'materialize-css';
 
-export default () => {
+const AuthenticatedHeader = ({ logout, history }) => {
   useEffect(() => {
     const dropDownElems = document.querySelectorAll('.dropdown-trigger');
     const sideNavElems = document.querySelectorAll('.sidenav');
     Materialize.Sidenav.init(sideNavElems);
     Materialize.Dropdown.init(dropDownElems);
   });
 
+  const logoutUser = () => {
+    logout(logoutAccount(history));
+  };
+
   return (
-    <Fragment>
+    <div>
```

Comment:
Any reason why the `Fragment` here was changed to a `div`?

Reply:
React Fragments group a list of children without adding extra nodes to the DOM.

This explains the error I was having- `Uncaught DOMException: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node`.

Replacing fragment with div solved my problem